R/fledgeling.R

```diff
@@ -0,0 +1,57 @@
+#' fledgeling object
+#'
+#' Contains the information that fledge manages about a repository.
+#' The internal structure of this object is subject to change.
+#'
+#' @param name The name of the package
+#' @param version A [package_version] that describes the current verson
+#' @param news A data frame
+#' @noRd
+new_fledgeling <- function(name, version, news) {
+  structure(
+    list(
+      name = name,
+      version = version,
+      news = news
+    ),
+    class = "fledgeling"
+  )
+}
+
+read_fledgeling <- function() {
+  package <- unname(desc::desc_get("Package"))
+  version <- desc::desc_get_version()
+
+  if (file.exists("NEWS.md")) {
+    news <- readLines("NEWS.md")
+  } else {
+    news <- character()
+  }
+
+  header_rx <- "^# +[a-zA-Z][a-zA-Z0-9.]*[a-zA-Z0-9] +(?<version>[0-9][0-9.-]*) *(?<date>.*)$"
+
+  first_level_headers <- grep(header_rx, news, perl = TRUE)
```

Comment:
I still think it'd be easier with tinkr :innocent: but understand the need for not using too many deps